src/helpers/pull-helpers/pull-request-parsing.ts

```diff
@@ -0,0 +1,105 @@
+import { encode } from "gpt-tokenizer";
+import { TokenLimits } from "../../types/llm";
+import { EncodeOptions } from "gpt-tokenizer/esm/GptEncoding";
+import { Context } from "../../types";
+
+export async function processPullRequestDiff(diff: string, tokenLimits: TokenLimits, logger: Context["logger"]) {
+  const { runningTokenCount, tokensRemaining } = tokenLimits;
+
+  // parse the diff into per-file diffs for quicker processing
+  const excludedFiles = ["bun.lockb"];
+  const perFileDiffs = parsePerFileDiffs(diff).filter((file) => !excludedFiles.includes(file.filename));
+
+  // quick estimate using a simple heuristic; 3.5 characters per token
```

Comment:
Just use the real tokenizer instead 

There's no point in doing this guess work when you can have 100% accurate results using the correct method.